File: 409.最长回文串.py
#
# @lc app=leetcode.cn id=409 lang=python
#
# [409] 最长回文串
#

# @lc code=start
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: int
        """
        a = set(s)
        b= list(a)
        oddc = 0
        flag = 0
        for i in range(len(b)):
            c = s.count(b[i])
            if c % 2 != 0:
                oddc += 1
                flag = 1
        l = len(s) - (oddc - flag)
        return l
        
# @lc code=end

File: 409.最长回文串.cpp
/*
 * @lc app=leetcode.cn id=409 lang=cpp
 *
 * [409] 最长回文串
 */

// @lc code=start
class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char, int> res;
        for(auto& ch : s)
            res[ch] += 1;
        int result = 0;
        int center = 0;
        for(auto item : res){
            if(item.second % 2 == 0)
                result += item.second;
            else
            {
                result += item.second - 1;
                center = 1;
            }
            
        }
        return result + center;


    }
};
// @lc code=end

